plugins/experimental/ssl_session_reuse/src/config.cc

```diff
@@ -62,7 +62,9 @@ Config::loadConfig(const std::string &filename)
     size_t n = info.st_size;
     std::string config_data;
     config_data.resize(n);
-    read(fd, const_cast<char *>(config_data.data()), n);
+    if (read(fd, const_cast<char *>(config_data.data()), n) < 0) {
```

Comment:
Out of paranoia I would change this to:
```
if (read(fd, const_cast<char *>(config_data.data()), n) != n) {
```

Reply:
Don't you need to return false if the call to open fails above?

Don't you need to make sure the file is closed at all return / exception points?  You could use ts::PostScript.

Reply:
Seem like valid concerns.

Reply:
It is returning false, it's the variable that's called `success`.